### PR TITLE
`wasm-bindgen-test-runner` fix run in parallel in same workspace

### PR DESCRIPTION
**Pull request by lumasepa**
Fix for #3419

Maybe we could delete the temporary directory after finishing the test so the target directory doesn't grow infinitely. If requested I will implement it.

### CODE REVIEW

File: crates/cli/src/bin/wasm-bindgen-test-runner/main.rs [hunk 1]
```diff
@@ -44,6 +44,11 @@ fn main() -> anyhow::Result<()> {
         None => bail!("must have a file to test as first argument"),
     };
 
+    let wasm_file_to_test_name = wasm_file_to_test
+        .file_name()
+        .and_then(|s| s.to_str())
+        .ok_or_else(|| anyhow!("file to test is not a valid file, can't extract file name"))?;
```
Comment (daxpedda) [inline crates/cli/src/bin/wasm-bindgen-test-runner/main.rs:50]:
```suggestion
        .context("file to test is not a valid file, can't extract file name")?;
```
That should work, see https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.context.

**Review by daxpedda — CHANGES_REQUESTED** [review]
This looks good to me, but yes, folders should be deleted after finishing.

**Comment by daxpedda** [comment]
Actually I just confirmed that Cargo itself doesn't delete tests between builds or tests, but I still think we should delete the folders because `wasm-bindgen` doesn't do any fingerprinting and will regenerate the file anyway.

File: crates/cli/src/bin/wasm-bindgen-test-runner/main.rs [hunk 2]
```diff
@@ -32,6 +32,16 @@ enum TestMode {
     Worker { no_modules: bool },
 }
 
+struct TmpDirDeleteGuard {
+    path: PathBuf,
+}
+
+impl Drop for TmpDirDeleteGuard {
+    fn drop(&mut self) {
+        let _ = fs::remove_dir_all(&self.path);
```
Comment (daxpedda) [inline crates/cli/src/bin/wasm-bindgen-test-runner/main.rs:41]: Can we log the error here instead of discarding it?

**Review by daxpedda — CHANGES_REQUESTED** [review]

File: crates/cli/src/bin/wasm-bindgen-test-runner/main.rs (hunk 2 above)
Comment (lumasepa) [reply crates/cli/src/bin/wasm-bindgen-test-runner/main.rs:41]: `eprintln!` is fine?

Comment (lumasepa) [reply crates/cli/src/bin/wasm-bindgen-test-runner/main.rs:41]: I saw that you are using env_logger, will use log then.

File: crates/cli/src/bin/wasm-bindgen-test-runner/main.rs [hunk 3]
```diff
@@ -32,6 +33,18 @@ enum TestMode {
     Worker { no_modules: bool },
 }
 
+struct TmpDirDeleteGuard {
+    path: PathBuf,
+}
```
Comment (daxpedda) [inline crates/cli/src/bin/wasm-bindgen-test-runner/main.rs:38]: Small nit, let's make this a tuple struct.

**Review by daxpedda — CHANGES_REQUESTED** [review]
LGTM, small nit.

**Review by daxpedda — APPROVED** [review]
LGTM, waiting for CI.

**Pull request merged**